src/main/java/seedu/planner/model/course/MajorDescription.java

```diff
@@ -29,6 +43,30 @@ public MajorDescription(Major major, List<ModuleDescription> modules) {
         this.modules = modules;
     }
 
+    private static class MajorDescriptionRetriever {
```

Comment:
Can you add a comment here saying that this is adapted from `seedu.planner.model.ModuleInfo.ModuleInfoRetriever`?